contracts/FastBreak.cdc

```diff
@@ -0,0 +1,682 @@
+/*
+    Fast Break Game Contract
+    Author: Jeremy Ahrens jer.ahrens@dapperlabs.com
+*/
+
+import NonFungibleToken from 0xNFTADDRESS
+import TopShot from 0xTOPSHOTADDRESS
+
+/// Game & Oracle Contract for Fast Break
+///
+pub contract FastBreak: NonFungibleToken {
+
+    /// Contract events
+    ///
+    pub event ContractInitialized()
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+    pub event FastBreakRunCreated(
+        id: String,
+        name: String,
+        runStart: UInt64,
+        runEnd: UInt64,
+        fatigueModeOn: Bool
+    )
+    pub event FastBreakRunStatusChange(id: String, newStatus: String)
+    pub event FastBreakGameCreated(
+        id: String,
+        name: String,
+        fastBreakRunID: String,
+        isPublic: Bool,
+        submissionDeadline: UInt64,
+        numPlayers: UInt64
+    )
+    pub event FastBreakGameStatusChange(id: String, newStatus: String)
+    pub event FastBreakNFTBurned(id: UInt64, serialNumber: UInt64)
+    pub event FastBreakNFTMinted(
+        id: UInt64,
+        fastBreakGameID: String,
+        serialNumber: UInt64,
+        mintingDate: UInt64,
+        topShots: [UInt64],
+        mintedTo: Address
+    )
+    pub event FastBreakSubmissionSent(
+        wallet: Address,
+        submittedAt: UInt64,
+        fastBreakGameID: String,
+        topShots: [UInt64]
+    )
+    pub event FastBreakGameWinner(
+        wallet: Address?,
+        submittedAt: UInt64,
+        fastBreakGameID: String,
+        topShots: [UInt64]
+    )
+    pub event FastBreakGameStatAdded(
+        fastBreakGameID: String,
+        name: String,
+        type: String,
+        valueNeeded: UInt64
+    )
+
+    /// Named Paths
+    ///
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+    pub let OracleStoragePath:       StoragePath
+    pub let OraclePrivatePath:      PrivatePath
+
+    /// Contract variables
+    ///
+    pub var totalSupply:        UInt64
+
+    /// Metadata Dictionaries
+    ///
+    access(self) let fastBreakRunByID:        {String: FastBreakRun}
+    access(self) let fastBreakGameByID:           {String: FastBreakGame}
```

Comment:
Any reason the IDs are strings and not integers?

Reply:
Artifact of doing the off-chain version first. I used strings there.